docs/guides/js-waku/store-retrieve-messages.md

```diff
@@ -0,0 +1,219 @@
+	// Render the message/payload in your application
+    console.log(wakuMessage);
+};
+
+// Set the query options
+const queryOptions = {
+	pageSize: 5,
+};
+
+// Query the Store node
+await node.store.queryOrderedCallback(
+	[decoder],
+	callback,
+	queryOptions,
+);
+```
+
+### `queryGenerator`
+
+The `store.queryGenerator()` function provides more control and flexibility over processing messages retrieved from `Store` nodes through [Async Generators](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/AsyncGenerator). It accepts these parameters:
+
+- `decoders`: List of `decoders` that specify the `content topic` to query for and their [message decryption](https://rfc.vac.dev/spec/26/) methods.
+- `options` (optional): [Query options](/guides/js-waku/store-retrieve-messages#store-query-options) to filter the retrieved messages.
+
+```js
+// Set the query options
+const queryOptions = {
+	pageSize: 5,
+};
+
+// Create the store query
+const storeQuery = node.store.queryGenerator(
+	[decoder],
+	queryOptions,
+);
+
+// Process the messages
+for await (const messagesPromises of storeQuery) {
+	// Fulfill all the messages promises
+    const messages = await Promise.all(messagesPromises);
+    // Render the message/payload in your application
+    console.log(messages);
+}
+```
+
+## Store Query Options
+
+### `pageDirection`
+
+The `pageDirection` option specifies the direction in which pages are retrieved:
+
+- `BACKWARD` (default): Most recent page first.
+- `FORWARD`: Oldest page first.
+
+```js
+import { PageDirection } from "@waku/sdk";
+
+// Retrieve recent messages first
+const queryOptions = {
+	pageDirection: PageDirection.BACKWARD,
+};
+
+// Retrieve oldest messages first
+const queryOptions = {
+	pageDirection: PageDirection.FORWARD,
+};
+```
+
+:::info
+The `pageDirection` option does not affect the ordering of messages within the page, as the oldest message always returns first.
+:::
+
+### `pageSize`
+
+The `pageSize` option specifies the number of messages to be returned per page. For example, consider a query that retrieves `20` messages per page:
+
+```js
+const queryOptions = {
+	pageSize: 20,
+};
+```
+
+### `timeFilter`
+
+The `timeFilter` option specifies a time frame to retrieve messages from. For example, consider a query that retrieves messages from the previous week:
+
+```js
+// Get the time frame
+const endTime = new Date();
+const startTime = new Date();
+startTime.setDate(endTime.getDate() - 7);
+
+// Retrieve a week of messages
+const queryOptions = {
+	timeFilter: {
+		startTime,
+		endTime,
+	},
+};
+```
+
+:::info
+If you omit the `timeFilter` option, the query will start from the beginning or end of the history, depending on the [page direction](#pagedirection).
+:::
+
+### `cursor`
+
+The `cursor` option specifies the starting index for retrieving messages. For example, consider a query that retrieves the first `10` messages and then continues with the next `10` messages:
```

Comment:
The current docs.waku.org page is incorrect.

```
import { waku } from "@waku/sdk";

// Create the callback function
const messages = [];
const callback = (wakuMessage) => {
    messages.push(wakuMessage);
// ADD return true; HERE and it will stop at the first page.
};

// Retrieve the first page of messages
// THIS RETRIEVES ALL MESSAGES if `return true` is not present
await node.store.queryOrderedCallback(
    [decoder],
    callback,
);

// Create the cursor
const lastMessage = messages[messages.length - 1];
const cursor = await waku.createCursor(lastMessage);

// Retrieve the next page of messages
// The message at the cursor index is excluded from the result
await node.store.queryOrderedCallback(
    [decoder],
    callback,
    {
        cursor: cursor,
    },
);
console.log(messages);
```